app/src/main/java/org/eyeseetea/malariacare/domain/usecase/push/PushUseCase.java

```diff
@@ -75,50 +78,13 @@ public void execute(final Callback callback) {
 
     @Override
     public void run() {
-
-        if (mPushController.isPushInProgress()) {
-            notifyPushInProgressError();
-            return;
-        }
-
-        mPushController.changePushInProgress(true);
-
-        try {
-            configureBanOrgUnitChangeListener();
-
-            boolean isBanned = isOrgUnitBanned();
-
-            if (isBanned) {
-                mPushController.changePushInProgress(false);
-            } else {
-                runPush();
-            }
-
-        } catch (Exception e) {
-            mPushController.changePushInProgress(false);
-            notifyPushError();
-        }
-    }
-
-    private void configureBanOrgUnitChangeListener() {
-        mOrganisationUnitRepository.setBanOrgUnitChangeListener(
-                new IOrganisationUnitRepository.BanOrgUnitChangeListener() {
-                    @Override
-                    public void onBanOrgUnitChanged(OrganisationUnit organisationUnit) {
-                        if (organisationUnit.isBanned()) {
-                            notifyBannedOrgUnitError();
-                        } else {
-                            notifyReOpenOrgUnit();
-                        }
-                    }
-                });
+        mPushUseCaseStrategy.run();
     }
 
-    private boolean isOrgUnitBanned() {
+    public boolean isOrgUnitBanned() throws NetworkException, ApiCallException {
```

Comment:
is dangerous refactor to public this methods for be invokes from strategy but can be invoked from presentation layer also, in PushUseCase only execute method should be public then I think is better refactor for move this methods to strategy base